File: archives/python/4375.py
a = []
while 1:
	try:
		a.append(int(input()))
	except EOFError:
		break

for num in a:
	i = 1
	while True:
		if (i % num == 0):
			print(len(str(i)))
			break
		else:
			i = i * 10 + 1

File: archives/cpp/4375.cpp
#include <bits/stdc++.h>
#define FOR(i, start, end) for (int i = start; i < end; i++)
#define endl "\n"
#define fastio ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
#pragma GCC optimize("Ofast")
using namespace std;

int main() {
    fastio

    while (true) {
        int n; cin >> n;
        if (cin.eof()) { break; }

        int num = 0;
        int cnt = 1;
        while (true) {
            num = num*10 + 1;
            num %= n;
            if (num == 0) {
                cout << cnt << endl;
                break;
            }
            cnt++;
        }
    }

    return 0;
}